src/Fixer/Phpdoc/PhpdocTypesFixer.php

```diff
@@ -129,32 +128,18 @@ public function getPriority(): int
 
     protected function normalize(string $type): string
     {
-        $typeExpression = new TypeExpression($type, null, []);
-
-        $typeExpression->walkTypes(function (TypeExpression $type): void {
-            if (!$type->isUnionType()) {
-                $value = $type->toString();
-                $valueLower = strtolower($value);
-                if (isset($this->typesSetToFix[$valueLower])) {
-                    $value = $valueLower;
-                }
-
-                // normalize shape/callable/generic identifiers too
-                // TODO parse them as inner types and this will be not needed then
-                $value = Preg::replaceCallback(
-                    '/^(\??\s*)([^()[\]{}<>\'"]+)(?<!\s)(\s*[\s()[\]{}<>])/',
-                    fn ($matches) => $matches[1].$this->normalize($matches[2]).$matches[3],
-                    $value
-                );
-
-                // TODO TypeExpression should be immutable and walkTypes method should be changed to mapTypes method
-                \Closure::bind(static function () use ($type, $value): void {
-                    $type->value = $value;
-                }, null, TypeExpression::class)();
-            }
-        });
-
-        return $typeExpression->toString();
+        $typeLower = strtolower($type);
+        if (isset($this->typesSetToFix[$typeLower])) {
+            $type = $typeLower;
+        }
+
+        // normalize shape/callable/generic identifiers too
+        // TODO parse them as inner types and this will be not needed then
+        return Preg::replaceCallback(
+            '/^(\??\s*)([^()[\]{}<>\'"]+)(?<!\s)(\s*[\s()[\]{}<>])/',
```

Comment:
I belive this regex cannot fix unioned/intersected type thus walkTypes is needed and test should be added.